### PR TITLE
Changes to coverage handling

### PR DESCRIPTION
**Pull request by jskeet**
- Add --coverage flag to build.sh instead of having a separate script
- Use --coverage from appveyor.sh
- In Kokoro integration tests, detect the presence of a codecov-token file from keystore and execute coverage based on that

(This is an alternative to #2540.)

### CODE REVIEW

File: .kokoro/runintegrationtests.sh [hunk 1]
```diff
@@ -11,10 +25,28 @@ cd ..
...
+
+if [[ -f "$KOKORO_KEYSTORE_DIR/73609_codecov-token" ]]
+then
+  CODECOV_TOKEN=$(cat "$KOKORO_KEYSTORE_DIR/73609_codecov-token")
+  SCRIPT_FLAGS=--coverage
+  choco install codecov
+fi
+
+# Build the libraries and run unit tests, optionally with coverage.
+./build.sh $BUILD_FLAGS
```
Comment (amanda-tarafa) [inline .kokoro/runintegrationtests.sh]: $SCRIPT_FLAGS?

File: .kokoro/runintegrationtests.sh [hunk 2]
```diff
@@ -2,6 +2,20 @@
...
+# Function used to upload coverage, if CODECOV_TOKEN is set.
+# $1 is used as the flag to codecov to indicate the report name.
+maybe_upload_coverage() {
+  if [[ -z "$CODECOV_TOKEN" ]]
+  then
+    return 0
+  fi
+  
+  ./createcoveragereport.sh
+  # Assume we've created the coverage file by this point. If we haven't, there shuold already have been an error.
```
Comment (amanda-tarafa) [inline .kokoro/runintegrationtests.sh]: nit: "should already"

Comment (jskeet) [reply .kokoro/runintegrationtests.sh]: Done.

File: .kokoro/runintegrationtests.sh (hunk 1 above)
Comment (jskeet) [reply .kokoro/runintegrationtests.sh]: Doh! Thanks :)

File: .kokoro/runintegrationtests.sh [hunk 3]
```diff
@@ -2,6 +2,20 @@
...
+# $1 is used as the flag to codecov to indicate the report name.
+maybe_upload_coverage() {
+  if [[ -z "$CODECOV_TOKEN" ]]
+  then
+    return 0
+  fi
+  
+  ./createcoveragereport.sh
+  # Assume we've created the coverage file by this point. If we haven't, there should already have been an error.
+  codecov -f "coverage/coverage-filtered.xml" --flag $1 --c $KOKORO_GITHUB_COMMIT --b KOKORO_BUILD_NUMBER
```
[the remark this answers is not in the document]
Comment (jskeet) [reply .kokoro/runintegrationtests.sh]: Humbug - not sure how that happened. Thanks!

**Pull request merged**